src/containers/SinglePost.js

```diff
@@ -24,24 +24,32 @@ import AccessErrorMessage from '../components/AccessErrorMessage'
 import CoverImagePage from '../components/CoverImagePage'
 import EventPost from '../components/EventPost'
 import ProjectPost from '../components/ProjectPost'
+import ProjectActivityCard from '../components/ProjectActivityCard'
 import { getCurrentCommunity } from '../models/community'
 import { denormalizedPost, getComments, getPost, isMessageThread } from '../models/post'
 import { fetch, ConnectedPostList } from './ConnectedPostList'
 const { array, bool, object, string, func } = React.PropTypes
 
-const subject = 'community'
-
-const showTaggedPosts = post =>
-  post.tag && post.type === 'project'
+const subject = 'post'
 
 @prefetch(({ store, dispatch, params: { id }, query }) =>
   dispatch(fetchPost(id))
+  .then(action => {
+    if (action.payload.parent_post_id) {
+      return dispatch(fetchPost(action.payload.parent_post_id))
+        .then(() => Promise.resolve(action))
```

Comment:
Promise.resolve is redundant here; `.then(() => action)` suffices. Actually it's redundant two lines below as well since you're already inside a `then`.

Also, check for `action.error` and bail out early before assuming `action.payload.parent_post_id` exists.